src/swigpytest.cc

```diff
@@ -0,0 +1,137 @@
+//  Copyright (c) 2021-22, VMware Inc, and the Certifier Authors.  All rights
+//  reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ****************************************************************************
+// swigpytest.cc : To test swigpytest.i SWIG interfaces.
+// These are just dummy class definitions, based on the stuff defined in
+// certifier_framework.h . See swigpytest.h for details.
+// ****************************************************************************
+
+#include <string.h>
+#include "swigpytest.h"
+
+using namespace swigpytests;
+
+// ****************************************************************************
+swigpytests::cc_trust_data::cc_trust_data() {
+  serialized_policy_cert_ = "Unknown-root-cert";
+}
+
+swigpytests::cc_trust_data::~cc_trust_data() {}
+
+// ****************************************************************************
+swigpytests::secure_authenticated_channel::secure_authenticated_channel() {
+  role_ = "Undefined-role";
+}
+
+swigpytests::secure_authenticated_channel::secure_authenticated_channel(
+    string &role) {
+  role_ = role;
+}
+
+swigpytests::secure_authenticated_channel::~secure_authenticated_channel() {}
+
+bool swigpytests::secure_authenticated_channel::init_client_ssl(
+    const string &asn1_root_cert) {  // In
+
+  asn1_root_cert_ = asn1_root_cert;
+  swig_wrap_fn_name_.assign("init_client_ssl-const-string-asn1_root_cert");
```

Comment:
This member field provides a testing-hook.

Each method that is invoked thru these SWIG wrappers will store a bread-crumb in this field, indicating which method was actually executed.

The python pytest code will cross-check this and assert the expected method's bread-crumb string.

If ever, due to changes in SWIG typemapping rules, if the flow is changed, the pytest will stumble upon this assert failing, which will help our debugging.

Reply:
Does it make sense to put this file into "tests/pytests"?

Reply:
@yelvmw - The 'problem' with that is that this file has to be linked with the generated `wrap*.cc` file. We will need to write a new Makefile in `tests/pytests` to compile these files. Or have the `certifier.mak` reference the dir paths in `tests/pytests` etc.

There are other files such as `certifier_tests.cc` in this `src/` dir, so may be I thought it would be ok for the short-term to leave this file in this dir. However, maybe I should rename it as `swigpytests.cc` for consistency with other `*tests.cc` files.

A related issue is the location of `swigpytest.h` and `swigpytest.i` in include/ dir. That's messy too, and I wasn't too happy with that. 

How about I try to do this:

- Create under `tests/pytests` two dirs: `src/` and `include/`
- Move the .h and .i files to this new `include/` dir.
- Move the `swigpytests.cc` to this new `src/` dir

Add a new Makefile to this area ... and rework the build commands.

If you are agreeable with some approach on these lines, I'll rework stuff. Won't know how things work out till I try.

Reply:
I see the concern now. I'll leave it to you then. Your plan about src/ and include/ in pytests sounds good to me. But it turned out to be a big hassle, I can live with the current solution too. Thanks!

Reply:
@yelvmw -- I reworked the files in this change set to be `swigpytests.cc`, `swigpytests.h` etc.

The rearrangement of these files to live inside `tests/pytests` will will take time.

Also see my [previous comment](https://github.com/vmware-research/certifier-framework-for-confidential-computing/pull/182#discussion_r1309524416) with my thoughts on this file-location aspect.

Let's wait for this area to settle down ... and then will rework.